ingenialink/ethercat/network.py

```diff
@@ -153,12 +166,60 @@ def disconnect_from_slave(self, servo: EthercatServo) -> None:  # type: ignore [
 
         """
         servo.stop_status_listener()
+        if not self._change_drive_state(servo.slave, pysoem.INIT_STATE):
+            logger.warning("Drive can not reach Init state")
         self.servos.remove(servo)
         if not self.servos:
             self.stop_status_listener()
             self._ecat_master.close()
             self.__is_master_running = False
 
+    def start_pdos(self) -> None:
+        """Configure the PDOs and set slave state to OP for all slaves with mapped PDOs"""
+        self._ecat_master.config_map()
+        self._ecat_master.state = pysoem.SAFEOP_STATE
+        op_servo_list: List[EthercatServo] = []
+        for servo in self.servos:
+            if not (servo._rpdo_maps or servo._tpdo_maps):
+                continue
+            op_servo_list.append(servo)
+            if not self._change_drive_state(servo.slave, pysoem.SAFEOP_STATE):
+                logger.warning("Drive can not reach SafeOp state")
+        if not op_servo_list:
+            logger.warning("No drives has PDO mapping")
+            return
+        # TODO Add porcessdata function INGK-799
+        self._ecat_master.send_processdata()
+        porcessdata_wkc = self._ecat_master.receive_processdata(
+            timeout=self.ECAT_PROCESSDATA_TIMEOUT
+        )
+        for servo in op_servo_list:
+            self._change_drive_state(servo.slave, pysoem.OP_STATE)
+
+    def stop_pdos(self) -> None:
+        """For all slaves in OP or SafeOp state, set state to PreOp"""
+        self._ecat_master.read_state()
+        for servo in self.servos:
+            if servo.slave.state not in [pysoem.OP_STATE, pysoem.SAFEOP_STATE]:
+                continue
+            if not self._change_drive_state(servo.slave, pysoem.PREOP_STATE):
+                logger.warning("Drive can not reach PreOp state")
+
+    def _change_drive_state(self, slave: "CdefSlave", target_state: int) -> bool:
```

Comment:
In general I would avoid references to "drive", "servo"... Specially for new API or internal code. 
Node is a better alternative. And is also more clear for multiaxis.
This is a communications lib that could be potentially used for other products.

Reply:
We have a problem with that, we use drive, servo, slave, and node indistinctly. Initially, the idea was use drive in ingenialink and servo in ingeniamotion, but I think we don't always meet this criteria.

Reply:
servo makes kind of sense in ingeniamotion, but here we should stick to slave or node if possible

Reply:
88889db